stm32-modules/include/flex-stacker/flex-stacker/tof_sensor_task.hpp

```diff
@@ -173,17 +178,122 @@ class TOFSensorTask {
+            response = messages::AcknowledgePrevious{
+                .responding_to_id = m.id,
+                .with_error = errors::ErrorCode::TMF8820_MEASURE_ERROR};
+            return send_response(response);
+        }
+
+        // Success, Set state and send response
+        sensor->state = MEASURING;
+        sensor->message_id = 0;
+        response = messages::ManageTOFMeasurementResponse{
+            .responding_to_id = m.id,
+            .sensor_id = m.sensor_id,
+            .kind = m.kind,
+            .len = (uint16_t)len,
+        };
+        send_response(response);
+    }
+
+    auto visit_message(const messages::GetTOFMeasurementMessage& m) -> void {
+        messages::HostCommsMessage response;
+        auto sensor = &get_sensor(m.sensor_id);
+        if (!sensor->ok || sensor->state != MEASURING) {
+            reset_measurement_state(m.sensor_id);
+            response = messages::AcknowledgePrevious{
+                .responding_to_id = m.id,
+                .with_error = errors::ErrorCode::TMF8820_MEASURE_ERROR};
+            return send_response(response);
+        }
+
+        // Resend previous chunk if requested and there is one
+        if (m.resend && sensor->message_id > 0) {
+            response = messages::GetTOFMeasurementResponse{
+                .responding_to_id = m.id,
+                .sensor_id = m.sensor_id,
+                .id = sensor->message_id,
+                .data = sensor->current_frame.data(),
+            };
+            return send_response(response);
+        }
+
+        // Get the next histogram chunk
+        auto ret = sensor->driver.get_histogram_chunk(m.sensor_id,
+                                                      sensor->current_frame);
+        if (ret == HIST_ERROR) {
+            reset_measurement_state(m.sensor_id);
+            response = messages::AcknowledgePrevious{
+                .responding_to_id = m.id,
+                .with_error = errors::ErrorCode::TMF8820_MEASURE_ERROR};
+            return send_response(response);
+        }
+
+        // Send histogram chunk
+        if (ret == HIST_OK || ret == HIST_DONE) {
+            sensor->message_id += 1;
+            response = messages::GetTOFMeasurementResponse{
+                .responding_to_id = m.id,
+                .sensor_id = m.sensor_id,
+                .id = sensor->message_id,
+                .data = sensor->current_frame.data(),
+            };
+            // This is the last packet, stop measurement and reset state
+            if (ret == HIST_DONE) {
+                reset_measurement_state(m.sensor_id);
+            }
+            return send_response(response);
+        }
+    }
+
+    auto send_response(messages::HostCommsMessage response) -> void {
         static_cast<void>(_task_registry->send_to_address(
             response, Queues::HostCommsAddress));
     }
 
+    auto reset_measurement_state(TOFSensorID sensor_id) -> void {
+        auto sensor = &get_sensor(sensor_id);
```

Comment:
Why take the address and pointerfy this? it returns a reference, so this doesn't do much

Reply:
cause Im a dummy, thanks for pointing it out.